astropy/coordinates/representation/base.py

```diff
@@ -898,16 +974,12 @@ def with_differentials(self, differentials):
         if not differentials:
             return self
 
-        args = [getattr(self, component) for component in self.components]
-
-        # We shallow copy the differentials dictionary so we don't update the
-        # current object's dictionary when adding new keys
-        new_rep = self.__class__(
-            *args, differentials=self.differentials.copy(), copy=False
+        differentials = self.differentials | self._validate_differentials(differentials)
+        return self.__class__(
+            *[getattr(self, component) for component in self.components],
+            differentials=self.differentials | differentials,
+            copy=False,
```

Comment:
```suggestion
        return self.__class__(
            *[getattr(self, component) for component in self.components],
            differentials=self.differentials | self._validate_differentials(differentials),
            copy=False,
```

Reply:
I'm fairly sure `ruff` will spread this over more lines...

Reply:
If id doesn't then it should be `differentials=differentials`

Reply:
Oops... thanks!

Reply:
Ruff does split it over two lines, which feels less clear. I ended up going with removing the `self.differentials` on l.977 - separate out validating from merging seemed most logical, giving a clearer traceback if the validation fails.

I'll wait a bit for further comments -- here or on #17016 -- before pushing this up.